addons/l10n_rs_edi/models/account_move.py

```diff
@@ -0,0 +1,164 @@
+        comodel_name='ir.attachment',
+        string="eFaktura XML Attachment",
+        compute=lambda self: self._compute_linked_attachment_id('l10n_rs_edi_attachment_id', 'l10n_rs_edi_attachment_file'),
+        depends=['l10n_rs_edi_attachment_file'],
+    )
+
+    l10n_rs_edi_state = fields.Selection(
+        string="Serbia E-Invoice state",
+        selection=[
+            ('sent', 'Sent'),
+            ('sending_failed', 'Error'),
+        ],
+        tracking=True,
+        readonly=True,
+        copy=False,
+    )
+
+    l10n_rs_edi_error = fields.Text(
+        string="Serbia E-Invoice error",
+        copy=False,
+        readonly=True,
+    )
+
+    l10n_rs_tax_date_obligations_code = fields.Selection(
+        string="Tax Date Obligations",
+        selection=[
+            ('35', 'By Delivery Date'),
+            ('3', 'By Issuance Date'),
+            ('432', 'By Billing System'),
+        ],
+        store=True,
+        readonly=False,
+        compute='_compute_l10n_rs_tax_date_obligations_code',
+    )
+    l10n_rs_edi_invoice = fields.Char("Invoice Id", copy=False)
+    l10n_rs_edi_sales_invoice = fields.Char("Sales Invoice Id", copy=False)
+    l10n_rs_edi_purchase_invoice = fields.Char("Purchase Invoice Id", copy=False)
+
+    @api.depends('country_code', 'move_type')
+    def _compute_show_delivery_date(self):
+        # EXTENDS 'account'
+        super()._compute_show_delivery_date()
+        for move in self:
+            if move.l10n_rs_edi_is_eligible:
+                move.show_delivery_date = True
+
+    @api.depends("country_code", "move_type")
+    def _compute_l10n_rs_edi_is_eligible(self):
+        for move in self:
+            move.l10n_rs_edi_is_eligible = move.country_code == 'RS' and move.is_sale_document()
+
+    @api.depends('country_code')
+    def _compute_l10n_rs_tax_date_obligations_code(self):
+        for move in self:
+            if move.country_code == 'RS':
+                move.l10n_rs_tax_date_obligations_code = '3'
+
+    @api.depends('l10n_rs_edi_state')
+    def _compute_show_reset_to_draft_button(self):
+        # EXTENDS 'account'
+        super()._compute_show_reset_to_draft_button()
+        for move in self:
+            if move.show_reset_to_draft_button and move.l10n_rs_edi_state == 'sent':
+                move.show_reset_to_draft_button = False
+
+    @api.depends('l10n_rs_edi_is_eligible')
+    def _compute_l10n_rs_edi_uuid(self):
+        for move in self:
+            if move.l10n_rs_edi_is_eligible and not move.l10n_rs_edi_uuid:
+                move.l10n_rs_edi_uuid = uuid.uuid4()
+
+    def button_draft(self):
+        # EXTENDS 'account'
+        self.write({
+            "l10n_rs_edi_error": False,
+            "l10n_rs_edi_state": False,
+        })
+        return super().button_draft()
+
+    def _l10n_rs_edi_send(self, xml, sendToCir):
+        self.ensure_one()
+        self.env['res.company']._with_locked_records(self)
```

Comment:
Lock all the records at once before calling the method. This is doing an sql query